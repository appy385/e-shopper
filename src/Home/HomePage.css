.fruits-vegetables-row {
    display: flex;
    margin: 10px;
    flex-wrap: wrap;
}

@media screen and (max-width: 24rem){
    .fruits-vegetables-row {
        flex-direction: column;
    }
}