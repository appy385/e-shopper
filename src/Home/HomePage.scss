.home {
    margin: 1rem;
}

.home-category {
    padding: 0 0 0.5rem 0;
}
.home-category-row {
    display: flex;
    flex-wrap: wrap;
}

@media screen and (max-width: 24rem){
    .home-category-row  {
        justify-content: center;
    }
    .home-category {
        text-align: center;
    }
}