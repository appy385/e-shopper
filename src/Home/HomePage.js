import React from 'react';
import PropTypes from 'prop-types';
import Card from './Card';

import './HomePage.scss';

const HomePage = ({ products, onIncrement, onDecrement }) => (
  <div className="home">
    <div className="home-page">
      {
            Object.entries(products).map((categoryProducts) => (
              <div className="home-category" key={categoryProducts[0]}>
                <strong>{categoryProducts[0]}</strong>
                <div className="home-category-row" data-testid="home-category-row">
                  { categoryProducts[1].map((product) => (
                    <Card
                      key={product.id}
                      product={product}
                      onIncrement={() => onIncrement(product.id, product.category)}
                      onDecrement={() => onDecrement(product.id, product.category)}
                    />
                  ))}
                </div>
              </div>
            ))
        }
    </div>
  </div>
);

const productShape = PropTypes.shape({
  id: PropTypes.number,
  name: PropTypes.string,
  quantity: PropTypes.number,
  price: PropTypes.number,
  url: PropTypes.string,
});
HomePage.propTypes = {
  products: PropTypes.objectOf(PropTypes.arrayOf(productShape)).isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,

};

export default HomePage;
