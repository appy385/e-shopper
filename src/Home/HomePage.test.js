import React from 'react';
import { render, screen } from '@testing-library/react';
import HomePage from './HomePage';

describe(HomePage.name, () => {
  let mockProducts;
  let mockInc;
  let mockDec;
  beforeEach(() => {
    mockProducts = [{
      id: 1,
      src: 'https://images.pexels.com/photos/61127/pexels-photo-61127.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500',
      seller: 'Fresho',
      productName: 'Banana-Robusta',
      quantity: '1kg',
      price: 40,
      count: 0,
    }];
    mockInc = jest.fn();
    mockDec = jest.fn();
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should match snapshot for Card component', () => {
    const { container } = render(<HomePage
      products={mockProducts}
      onIncrement={mockInc}
      onDecrement={mockDec}
    />);
    expect(container).toMatchSnapshot();
  });

  test('should display the product category "Fruits and Vegetables"', () => {
    render(<HomePage
      products={mockProducts}
      onIncrement={mockInc}
      onDecrement={mockDec}
    />);
    screen.getByText('Fruits and Vegetables');
  });
});
