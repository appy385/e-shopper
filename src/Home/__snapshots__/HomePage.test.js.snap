// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should match snapshot for Card component 1`] = `
<div>
  <div
    class="home"
  >
    <div
      class="home-page"
    >
      <div
        class="home-category"
      >
        <strong>
          Fruits and Vegetables
        </strong>
      </div>
      <div
        class="fruits-vegetables-row"
      >
        <div
          class="product-card"
        >
          <img
            alt="product"
            src="https://images.pexels.com/photos/61127/pexels-photo-61127.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500"
          />
          <div
            class="product-seller"
          >
            Fresho
          </div>
          <div
            class="product-name"
          >
            Banana-Robusta
          </div>
          <div
            class="product-quantity"
          >
            1kg
          </div>
          <div
            class="product-card-box"
          >
            <div
              class="product-price"
            >
              MRP: 40 /-
            </div>
            <div
              class="quantity-container"
            >
              <button
                class="quantity-inc-button"
                type="button"
              >
                +
              </button>
              <span
                class="quantity-basket-count"
              >
                0 in Basket
              </span>
              <button
                class="quantity-dec-button"
                type="button"
              >
                -
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
